.github/workflows/test-rke2.yaml

```diff
@@ -0,0 +1,108 @@
+
+      - name: Checkout repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502 # v4
+        with:
+          role-to-assume: ${{ secrets.AWS_COMMERCIAL_ROLE_TO_ASSUME }}
+          role-session-name: ${{ github.job || github.event.client_payload.pull_request.head.sha || github.sha }}
+          aws-region: ${{ env.UDS_REGION }}
+          role-duration-seconds: 21600
+
+      - name: Environment setup
+        uses: ./.github/actions/setup
+        with:
+          registry1Username: ${{ secrets.IRON_BANK_ROBOT_USERNAME }}
+          registry1Password: ${{ secrets.IRON_BANK_ROBOT_PASSWORD }}
+          ghToken: ${{ secrets.GITHUB_TOKEN }}
+          chainguardIdentity: ${{ secrets.CHAINGUARD_IDENTITY }}
+
+      # - name: Install eksctl
+      #   run: uds run -f tasks/iac.yaml install-eksctl --no-progress
+
+      - name: Setup Tofu
+        uses: opentofu/setup-opentofu@12f4debbf681675350b6cd1f0ff8ecfbda62027b # v1.0.4
+        with:
+          tofu_version: 1.8.2
+          tofu_wrapper: false
+
+      - name: Create UDS Core Package
+        run: ZARF_ARCHITECTURE=amd64 uds run -f tasks/create.yaml standard-package --no-progress --set FLAVOR=${{ inputs.flavor }} --set K8S_DISTRO=rke2
+
+      - name: Create Core Bundle
+        run: uds create .github/bundles/rke2 --confirm
+
+      - name: Create IAC
+        run: uds run -f tasks/iac.yaml create-iac --no-progress --set K8S_DISTRO=rke2
+        timeout-minutes: 20
+
+      - name: Get Kubeconfig
+        run: uds run -f tasks/iac.yaml rke2-get-kubeconfig --no-progress
+        timeout-minutes: 20
+
+      - name: Wait for RKE2 Cluster
+        run: uds run -f tasks/iac.yaml rke2-cluster-ready --no-progress
+        env:
+          KUBECONFIG: ".github/test-infra/aws/rke2/rke2-config"
+        timeout-minutes: 20
+
+      - name: Deploy Core Bundle
+        env:
+          KUBECONFIG: ".github/test-infra/aws/rke2/rke2-config"
+          UDS_CONFIG: .github/bundles/rke2/uds-config.yaml
+        run: uds deploy .github/bundles/rke2/uds-bundle-uds-core-rke2-nightly-*.tar.zst --confirm
+        timeout-minutes: 60
+
+      - name: Remove UDS Core
+        if: always()
+        run: uds remove .github/bundles/rke2/uds-bundle-uds-core-rke2-*.tar.zst --confirm
+        env:
+          KUBECONFIG: ".github/test-infra/aws/rke2/rke2-config"
+        timeout-minutes: 20
+        continue-on-error: true
+
+      - name: Remove IAC
+        if: always()
+        run: uds run -f tasks/iac.yaml destroy-iac --no-progress --set K8S_DISTRO=rke2
+        timeout-minutes: 10
+        continue-on-error: true
+
+      # - name: Teardown RKE2 cluster
+      #   if: always()
+      #   run: uds run -f tasks/iac.yaml destroy-cluster --no-progress --set K8S_DISTRO=rke2
+      #   timeout-minutes: 30
+      #   continue-on-error: true
```

Comment:
might be good to cleanup this commented out code - at first glance I got concerned thinking we were missing a teardown 😅